lib/helpers/mail/Mail.php

```diff
@@ -916,16 +916,28 @@ class Mail implements \jsonSerializable
 
     public function __construct($from = null, $subject = null, $to = null, $content = null)
     {
-        if (!empty($from) &&  !empty($subject) && !empty($to) && !empty($content))
-        {
+        $from = $from ?: null;
+        if ($from !== null) {
             $this->setFrom($from);
-            $personalization = new Personalization();
-            $personalization->addTo($to);
-            $this->addPersonalization($personalization);
+        }
+
+        $subject = $subject ?: null;
+        if ($subject !== null) {
             $this->setSubject($subject);
+        }
+
+        $content = $content ?: null;
+        if ($content !== null) {
             $this->addContent($content);
         }
 
+        $to = $to ?: null;
+        if ($to !== null) {
+            $personalization = new Personalization();
+            $personalization->addTo($to);
+
+            $this->addPersonalization($personalization);
+        }
```

Comment:
IMHO this PR doesn't make any sense, we shouldn't use setters since it breaks `OOP encapsulation`, so it would make more sense if we contribute to make the code better.

I think `from`, `to` and `content` should be mandatory, remove the `null` from the `constructor`.

Also, it's fine if you don't want to put a `subject` in your e-mail, that null ability is fine, so I would suggest the code to be like:

```php
public function __construct($from, $to, $content, $subject = null)
{
    $this->from = $from;
    $this->to = $to;
    $this->content = $this->addContent($content);
    $this->subject = $subject;
}

/**
 * @param Content $content
 * @return Mail
 */
private function addContent($content)
{
    $this->content = $content;
    $this->addPersonalization((new Personalization)->addTo($this->to))
    return $this;
}
```

Reply:
Thanks @shonjord! I will take this into consideration as we release [the next iteration of this SDK](https://github.com/sendgrid/sendgrid-php/issues/413).